examples/grpc-kotlin/src/main/kotlin/example/armeria/grpc/kotlin/HelloServiceImpl.kt

```diff
@@ -3,129 +3,112 @@ package example.armeria.grpc.kotlin
-        responseObserver.onCompleted()
+    override suspend fun hello(request: HelloRequest): HelloReply = withArmeriaContext {
+        // Make sure that current thread is request context aware
+        ServiceRequestContext.current()
+        buildReply(toMessage(request.name))
     }
 
-    override fun lazyHello(request: HelloRequest, responseObserver: StreamObserver<HelloReply>) {
-        // You can use the event loop for scheduling a task.
-        ServiceRequestContext.current().contextAwareEventLoop().schedule({
-            responseObserver.onNext(buildReply(toMessage(request.name)))
-            responseObserver.onCompleted()
-        }, 3, TimeUnit.SECONDS)
+    override suspend fun lazyHello(request: HelloRequest): HelloReply = withArmeriaContext {
+        delay(3000L)
+        ServiceRequestContext.current()
+        buildReply(toMessage(request.name))
     }
 
     /**
      * Sends a [HelloReply] using `blockingTaskExecutor`.
      *
      * @see [Blocking service implementation](https://line.github.io/armeria/server-grpc.html#blocking-service-implementation)
      */
-    override fun blockingHello(request: HelloRequest, responseObserver: StreamObserver<HelloReply>) {
-        // Unlike upstream gRPC-Java, Armeria does not run service logic in a separate thread pool by default.
-        // Therefore, this method will run in the event loop, which means that you can suffer the performance
-        // degradation if you call a blocking API in this method. In this case, you have the following options:
-        //
-        // 1. Call a blocking API in the blockingTaskExecutor provided by Armeria.
-        // 2. Set `GrpcServiceBuilder.useBlockingTaskExecutor(true)` when building your GrpcService.
-        // 3. Call a blocking API in the separate thread pool you manage.
-        //
-        // In this example, we chose the option 1:
-        ServiceRequestContext.current().blockingTaskExecutor().submit {
-            try { // Simulate a blocking API call.
-                Thread.sleep(3000)
-            } catch (ignored: Exception) { // Do nothing.
-            }
-            responseObserver.onNext(buildReply(toMessage(request.name)))
-            responseObserver.onCompleted()
+    override suspend fun blockingHello(request: HelloRequest): HelloReply = withArmeriaBlockingContext {
+        try { // Simulate a blocking API call.
+            Thread.sleep(3000)
+        } catch (ignored: Exception) { // Do nothing.
         }
+        // Make sure that current thread is request context aware
+        ServiceRequestContext.current()
+        buildReply(toMessage(request.name))
     }
 
     /**
      * Sends 5 [HelloReply] responses when receiving a request.
      *
      * @see lazyHello(HelloRequest, StreamObserver)
      */
-    override fun lotsOfReplies(request: HelloRequest, responseObserver: StreamObserver<HelloReply>) {
+    override fun lotsOfReplies(request: HelloRequest): Flow<HelloReply> {
```

Comment:
Does this need to be flow? I wouldn't expect a for loop in a coroutine to cause any differences compared to our normal `withArmeriaContext` methods

Reply:
To clarify, it doesn't matter so much whether it's a flow or not but was hoping `withArmeriaContext` applies to all the methods without having yet another pattern of `flowOn`

Reply:
I’ve tried but it was impossible to apply because `withContext` only could be applied to suspend function. This function is just return flow and not suspend :-(

Reply:
Or we can wrap `withArmeriaContext` inside of flow block.

Reply:
I see - does that mean we can have

```
flow {
  withArmeriaContext {
    for (i...)
```
?

Definitely not great but I guess a bit better than a new concept `flowOn`. And probably good evidence to add to our issue about getting better support for customizing the context globally :)

Reply:
> I see - does that mean we can have

Yes, `flow` takes suspend function.
https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/common/src/flow/Builders.kt#L49

> And probably good evidence to add to our issue about getting better support for customizing the context globally :)

That sounds good. 👍

Reply:
The test failed, the nested block does not work...
```kt
flow {
  withArmeriaContext {
    for (i...)
```

`emit()` should be called from dispatchers of flow block or it throws ISE.
https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/common/src/flow/Builders.kt#L37-L45

Reply:
Too bad :( Let's stick with flowOn then since best we can do